### PR TITLE
docs(typography): updates typography guidelines as part of content audit

### LINKED ISSUE #3694: Review and update typography guidelines

### PR DESCRIPTION
**Pull request by edonehoo**
closes #3694

### CODE REVIEW

**Comment by edonehoo** [comment]
A few comments/questions:

1. I want to move the "patternfly text styles" subsections to H3 so that they'll appear in the TOC, but it doesn't look great right now. Would it be a difficult change to adjust the css to adapt to this updated structure? Or are there objections to this change?

2. Additionally, to make these H3s I removed `title` from `TitleLevel` and want to be sure that's okay 👀 

3. There's some red code within the grid at the end of the file. Not sure if that needs to be updated, or if it's fine to ignore.

**Review by andrew-ronaldson — CHANGES_REQUESTED** [review]
Looks like there is a small spacing issue following these tables. Looks like a tag isn't getting closed somewhere. Screenshot 2023-09-25 at 10 39 59 AM

File: packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md [hunk 1]
```diff
@@ -11,109 +11,113 @@ import correct from './typography_correct_spacing.png';
...
+- **Font size CSS variable:** The CSS variable that controls the size of your text. You can customize type sizes in your application by modifying the values associated with these variables. In general, this is not advised, and should be done with caution, as changes to text style can have far reaching side effects on spacing within certain components.
 
+### Super hero heading (4xl)
+**Note:** Use only for extra large titles in cards or similar locations. Do not use for page titles.
 <TitleLevel
-  title="Super hero heading (4xl)"
   className="pf-v5-c-title pf-m-lg ws-title-level-heading"
   asGrid
-  note="Use only for extra large titles in cards or similar locations. Do not use for page titles"
   styleProps={styleProps.superHero} />
```
Comment (edonehoo) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md:37]:
```suggestion
<div class="--pf-v5-global--spacer--sm"
</div>
```

File: packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md [hunk 2]
```diff
@@ -11,109 +11,114 @@ import correct from './typography_correct_spacing.png';
...
+<div class="--pf-v5-global--spacer--sm"
```
Comment (edonehoo) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md:37]:
```suggestion
<div class="--pf-v5-global--spacer--sm">
```

File: packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md [hunk 3]
```diff
@@ -43,32 +43,37 @@ The following sections outline common text styles that are used in PatternFly ap
   className="pf-v5-c-title pf-m-lg ws-title-level-heading"
   asGrid
   styleProps={styleProps.hero} />
-
+<div class="--pf-v5-global--spacer--sm">
```
Comment (edonehoo) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md:46]:
```suggestion
<div class="--pf-v5-global--spacer--lg">
```

File: packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md [hunk 4]
```diff
@@ -43,32 +43,37 @@ The following sections outline common text styles that are used in PatternFly ap
...
+</div>
 ### First level heading (2xl, H1)
 **Note**: Only 1 first level/H1 heading should ever exist on a page.
 <TitleLevel
   className="pf-v5-c-title pf-m-lg ws-title-level-heading"
   asGrid
   styleProps={styleProps.first} />
-
+<div class="--pf-v5-global--spacer--sm">
```
Comment (edonehoo) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md:54]:
```suggestion
<div class="--pf-v5-global--spacer--lg">
```

File: packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md [hunk 5]
```diff
@@ -43,32 +43,37 @@ The following sections outline common text styles that are used in PatternFly ap
...
+</div>
 ### Second level heading (xl, H2)
 <TitleLevel
   className="pf-v5-c-title pf-m-lg ws-title-level-heading"
   asGrid
   styleProps={styleProps.second} />
-
+<div class="--pf-v5-global--spacer--sm">
```
Comment (edonehoo) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md:61]:
```suggestion
<div class="--pf-v5-global--spacer--lg">
```

File: packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md [hunk 6]
```diff
@@ -43,32 +43,37 @@ The following sections outline common text styles that are used in PatternFly ap
...
+</div>
 ### Third level heading (lg, H3)
 <TitleLevel
   className="pf-v5-c-title pf-m-lg ws-title-level-heading"
   asGrid
   styleProps={styleProps.third} />
-
+<div class="--pf-v5-global--spacer--sm">
```
Comment (edonehoo) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md:68]:
```suggestion
<div class="--pf-v5-global--spacer--lg">
```

File: packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md [hunk 7]
```diff
@@ -43,32 +43,37 @@ The following sections outline common text styles that are used in PatternFly ap
...
+</div>
 ### Fourth level heading (md, H4)
 <TitleLevel
   className="pf-v5-c-title pf-m-lg ws-title-level-heading"
   asGrid
   styleProps = {styleProps.fourth} />
-
+<div class="--pf-v5-global--spacer--sm">
```
Comment (edonehoo) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md:75]:
```suggestion
<div class="--pf-v5-global--spacer--lg">
```

File: packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md [hunk 8]
```diff
@@ -11,109 +11,119 @@ import correct from './typography_correct_spacing.png';
...
 
+### Super hero heading (4xl)
+**Note:** Use only for extra large titles in cards or similar locations. Do not use for page titles.
 <TitleLevel
-  title="Super hero heading (4xl)"
   className="pf-v5-c-title pf-m-lg ws-title-level-heading"
   asGrid
-  note="Use only for extra large titles in cards or similar locations. Do not use for page titles"
   styleProps={styleProps.superHero} />
+<div class="--pf-v5-global--spacer--sm">
```
Comment (edonehoo) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md:37]:
```suggestion
<div class="--pf-v5-global--spacer--lg">
```

**Comment by edonehoo** [comment]
@andrew-ronaldson it still looked wrong on my end, so I fiddled more, and it still doesn't look right. should the spacers method be working or do we just need to use <br>?

**Comment by evwilkin** [comment]
@edonehoo I believe you're fighting against some code built into our template for typography page which is causing the spacing issues.  I've adjusted the template slightly based on what it looks like we're going for now (mostly moving the **Note** just below the heading name) and put up a draft PR #3769 to see if that looks to fix the issues here.

Once the preview link is available there feel free to let me know how it looks and we can get this PR updated accordingly 👍

**Comment by edonehoo** [comment]
@evwilkin ah tysm that looks far better! Is it possible to also make each "X heading" heading an H3 so that they appear in the TOC? If not - I think the changes you made are fine & can be integrated

File: packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md [hunk 9]
```diff
@@ -11,109 +11,148 @@ import correct from './typography_correct_spacing.png';
...
+**Note:** Use only for extra large titles in cards or similar locations. Do not use for page titles.
 
 <TitleLevel
-  title="Super hero heading (4xl)"
   className="pf-v5-c-title pf-m-lg ws-title-level-heading"
   asGrid
-  note="Use only for extra large titles in cards or similar locations. Do not use for page titles"
   styleProps={styleProps.superHero} />
 
+<div class="--pf-v5-global--spacer--lg">
```
Comment (evwilkin) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md:40]:
```suggestion
<div class="pf-v5-u-mt-lg> 
```

File: packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md [hunk 10]
```diff
@@ -11,109 +11,148 @@ import correct from './typography_correct_spacing.png';
...
+
+**Note:** Use only for extra large titles in cards or similar locations. Do not use for page titles.
+
 <TitleLevel
-  title="Hero heading (3xl)"
   className="pf-v5-c-title pf-m-lg ws-title-level-heading"
   asGrid
-  note="Use only for extra large titles in cards or similar locations. Do not use for page titles"
   styleProps={styleProps.hero} />
+<div class="--pf-v5-global--spacer--lg">
```
Comment (evwilkin) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md:51]:
```suggestion
<div class="pf-v5-u-mt-lg">
```

File: packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md [hunk 11]
```diff
@@ -11,109 +11,148 @@ import correct from './typography_correct_spacing.png';
...
+
+**Note**: Only 1 first level/H1 heading should ever exist on a page.
 
 <TitleLevel
-  title="First level heading (2xl, h1)"
   className="pf-v5-c-title pf-m-lg ws-title-level-heading"
   asGrid
-  note="Only one first level/h1 heading should ever exist on a page"
   styleProps={styleProps.first} />
+<div class="--pf-v5-global--spacer--lg">
```
Comment (evwilkin) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md:62]:
```suggestion
<div class="pf-v5-u-mt-lg">
```

File: packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md [hunk 12]
```diff
@@ -11,109 +11,148 @@ import correct from './typography_correct_spacing.png';
...
+</div>
+
+### Second level heading (xl, H2)
 
 <TitleLevel
-  title="Second level heading (xl, h2)"
   className="pf-v5-c-title pf-m-lg ws-title-level-heading"
   asGrid
   styleProps={styleProps.second} />
+<div class="--pf-v5-global--spacer--lg">
```
Comment (evwilkin) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md:71]:
```suggestion
<div class="pf-v5-u-mt-lg">
```

File: packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md [hunk 13]
```diff
@@ -11,109 +11,148 @@ import correct from './typography_correct_spacing.png';
...
+</div>
+
+### Third level heading (lg, H3)
 
 <TitleLevel
-  title="Third level heading (lg, h3)"
   className="pf-v5-c-title pf-m-lg ws-title-level-heading"
   asGrid
   styleProps={styleProps.third} />
+<div class="--pf-v5-global--spacer--lg">
```
Comment (evwilkin) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md:80]:
```suggestion
<div class="pf-v5-u-mt-lg">
```

File: packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md [hunk 14]
```diff
@@ -11,109 +11,148 @@ import correct from './typography_correct_spacing.png';
...
+</div>
+
+### Fourth level heading (md, H4)
 
 <TitleLevel
-  title="Fourth level heading (md, h4)"
   className="pf-v5-c-title pf-m-lg ws-title-level-heading"
   asGrid
   styleProps = {styleProps.fourth} />
+<div class="--pf-v5-global--spacer--lg">
```
Comment (evwilkin) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md:89]:
```suggestion
<div class="pf-v5-u-mt-lg">
```

File: packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md [hunk 15]
```diff
@@ -11,109 +11,148 @@ import correct from './typography_correct_spacing.png';
...
+
+**Note:** This is the default text style for paragraphs, lists, tables, etc. Some components instead use Red Hat Text at 700 font weight, which is Red Hat Text Medium (such as alerts and navigation).
 
 <TitleLevel
-  title="Default body text"
   className="pf-v5-c-title pf-m-lg ws-title-level-heading"
   asGrid
-  note="This is the default text style for paragraphs, lists, tables, etc. Some components use RedHatText at 700 font weight, which is RedHatText Medium (such as alerts and navigation)"
   styleProps = {styleProps.body} />
+<div class="--pf-v5-global--spacer--lg">
```
Comment (evwilkin) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md:100]:
```suggestion
<div class="pf-v5-u-mt-lg">
```

File: packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md [hunk 16]
```diff
@@ -11,109 +11,148 @@ import correct from './typography_correct_spacing.png';
...
+</div>
+
+### Small text
 
 <TitleLevel
-  title="Small text"
   className="pf-v5-c-title pf-m-lg ws-title-level-heading"
   asGrid
   styleProps = {styleProps.small} />
+<div class="--pf-v5-global--spacer--lg">
```
Comment (evwilkin) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md:109]:
```suggestion
<div class="pf-v5-u-mt-lg">
```

File: packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md [hunk 17]
```diff
@@ -11,109 +11,148 @@ import correct from './typography_correct_spacing.png';
...
+</div>
+
+### Tiny text
 
 <TitleLevel
-  title="Tiny text"
   className="pf-v5-c-title pf-m-lg ws-title-level-heading"
   asGrid
   styleProps = {styleProps.tiny} />
+<div class="--pf-v5-global--spacer--lg">
```
Comment (evwilkin) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md:118]:
```suggestion
<div class="pf-v5-u-mt-lg">
```

File: packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md [hunk 18]
```diff
@@ -11,109 +11,148 @@ import correct from './typography_correct_spacing.png';
...
+
+**Note:** Used for code blocks.
 
 <TitleLevel
-  title="Code"
   className="pf-v5-c-title pf-m-lg ws-title-level-heading"
   asGrid
-  note="Used for code blocks "
   styleProps = {styleProps.code} />
+<div class="--pf-v5-global--spacer--lg">
```
Comment (evwilkin) [inline packages/documentation-site/patternfly-docs/content/design-guidelines/styles/typography/typography.md:129]:
```suggestion
<div class="pf-v5-u-mt-lg">
```

**Comment by evwilkin** [comment]
@edonehoo the TOC updates proved to be harder than they should!  I went ahead and pushed a few changes here to your PR, you had the right idea with the spacer but the className wasn't quite right so the spacing wasn't working correctly.

**Comment by andrew-ronaldson** [comment]
Thanks for the spacer fix Evan. I saw they were the variable names and got side tracked looking for the appropriate classes.

**Comment by edonehoo** [comment]
@evwilkin @andrew-ronaldson woo thank you both! I think this can be merged

**Review by evwilkin — APPROVED** [review]
LGTM 👏

**Pull request merged**